Test failures - TestLinprogIPSparseCholmod

When I run the test suite (from master) on my own computer I get a whole heap of test failures from `TestLinprogIPSparseCholmod`.
@mdhaber

```
_______________________________________________________________ TestLinprogIPSparseCholmod.test_docstring_example ________________________________________________________________
scipy/optimize/tests/test_linprog.py:266: in test_docstring_example
    options=self.options, method=self.method)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_____________________________________________________________ TestLinprogIPSparseCholmod.test_inequality_constraints _____________________________________________________________
scipy/optimize/tests/test_linprog.py:393: in test_inequality_constraints
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________ TestLinprogIPSparseCholmod.test_inequality_constraints2 _____________________________________________________________
scipy/optimize/tests/test_linprog.py:406: in test_inequality_constraints2
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
__________________________________________________________________ TestLinprogIPSparseCholmod.test_bounds_mixed __________________________________________________________________
scipy/optimize/tests/test_linprog.py:477: in test_bounds_mixed
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________ TestLinprogIPSparseCholmod.test_bounds_equal_no_presolve ____________________________________________________________
scipy/optimize/tests/test_linprog.py:512: in test_bounds_equal_no_presolve
    method=self.method, options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_________________________________________________________________ TestLinprogIPSparseCholmod.test_zero_column_1 __________________________________________________________________
scipy/optimize/tests/test_linprog.py:527: in test_zero_column_1
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_zero_row_1 ___________________________________________________________________
scipy/optimize/tests/test_linprog.py:555: in test_zero_row_1
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_zero_row_2 ___________________________________________________________________
scipy/optimize/tests/test_linprog.py:563: in test_zero_row_2
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_______________________________________________________________ TestLinprogIPSparseCholmod.test_singleton_row_eq_2 _______________________________________________________________
scipy/optimize/tests/test_linprog.py:611: in test_singleton_row_eq_2
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_______________________________________________________________ TestLinprogIPSparseCholmod.test_singleton_row_ub_2 _______________________________________________________________
scipy/optimize/tests/test_linprog.py:633: in test_singleton_row_ub_2
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_infeasible ___________________________________________________________________
scipy/optimize/tests/test_linprog.py:644: in test_infeasible
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_unbounded ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:667: in test_unbounded
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_____________________________________________________ TestLinprogIPSparseCholmod.test_unbounded_below_no_presolve_corrected ______________________________________________________
scipy/optimize/tests/test_linprog.py:679: in test_unbounded_below_no_presolve_corrected
    options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
________________________________________________________________ TestLinprogIPSparseCholmod.test_cyclic_recovery _________________________________________________________________
scipy/optimize/tests/test_linprog.py:731: in test_cyclic_recovery
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
__________________________________________________________________ TestLinprogIPSparseCholmod.test_cyclic_bland __________________________________________________________________
scipy/optimize/tests/test_linprog.py:748: in test_cyclic_bland
    method=self.method, options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_______________________________________________________________ TestLinprogIPSparseCholmod.test_nontrivial_problem _______________________________________________________________
scipy/optimize/tests/test_linprog.py:783: in test_nontrivial_problem
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_________________________________________________________________ TestLinprogIPSparseCholmod.test_lpgen_problem __________________________________________________________________
scipy/optimize/tests/test_linprog.py:796: in test_lpgen_problem
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
__________________________________________________________________ TestLinprogIPSparseCholmod.test_network_flow __________________________________________________________________
scipy/optimize/tests/test_linprog.py:817: in test_network_flow
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_________________________________________________________ TestLinprogIPSparseCholmod.test_network_flow_limited_capacity __________________________________________________________
scipy/optimize/tests/test_linprog.py:848: in test_network_flow_limited_capacity
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
______________________________________________________ TestLinprogIPSparseCholmod.test_simplex_algorithm_wikipedia_example _______________________________________________________
scipy/optimize/tests/test_linprog.py:859: in test_simplex_algorithm_wikipedia_example
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
__________________________________________________________________ TestLinprogIPSparseCholmod.test_enzo_example __________________________________________________________________
scipy/optimize/tests/test_linprog.py:877: in test_enzo_example
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_________________________________________________________________ TestLinprogIPSparseCholmod.test_enzo_example_b _________________________________________________________________
scipy/optimize/tests/test_linprog.py:895: in test_enzo_example_b
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_________________________________________________________ TestLinprogIPSparseCholmod.test_enzo_example_c_with_degeneracy _________________________________________________________
scipy/optimize/tests/test_linprog.py:907: in test_enzo_example_c_with_degeneracy
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_______________________________________________________ TestLinprogIPSparseCholmod.test_enzo_example_c_with_infeasibility ________________________________________________________
scipy/optimize/tests/test_linprog.py:933: in test_enzo_example_c_with_infeasibility
    method=self.method, options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
_____________________________________________________________ TestLinprogIPSparseCholmod.test_basic_artificial_vars ______________________________________________________________
scipy/optimize/tests/test_linprog.py:949: in test_basic_artificial_vars
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_5400 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1009: in test_bug_5400
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_6139 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1029: in test_bug_6139
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_6690 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1066: in test_bug_6690
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_7044 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1089: in test_bug_7044
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_7237 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1134: in test_bug_7237
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_8174 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1153: in test_bug_8174
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_8174_2 ___________________________________________________________________
scipy/optimize/tests/test_linprog.py:1181: in test_bug_8174_2
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_8561 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1198: in test_bug_8561
    method=self.method)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_8662 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1219: in test_bug_8662
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_10124 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1310: in test_bug_10124
    method=self.method, options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_10349 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1329: in test_bug_10349
    method=self.method, options=self.options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
___________________________________________________________________ TestLinprogIPSparseCholmod.test_bug_10466 ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1365: in test_bug_10466
    method=self.method, options=o)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
____________________________________________________________________ TestLinprogIPSpecific.test_solver_select ____________________________________________________________________
scipy/optimize/tests/test_linprog.py:1604: in test_solver_select
    res1 = linprog(c, A_ub=A, b_ub=b, method=self.method, options=options)
scipy/optimize/_linprog.py:548: in linprog
    postsolve_args=postsolve_args, **solver_options)
scipy/optimize/_linprog_ip.py:1087: in _linprog_ip
    OptimizeWarning, stacklevel=3)
E   scipy.optimize.optimize.OptimizeWarning: Sparse cholesky is only available with scikit-sparse. Setting `cholesky = False`
============================================ 38 failed, 1495 passed, 10 skipped, 8 deselected, 1 xfailed, 2 xpassed in 52.59 seconds =============================================


```

--- a/scipy/optimize/_linprog_ip.py
+++ b/scipy/optimize/_linprog_ip.py
@@ -28,6 +28,7 @@ from ._linprog_util import _postsolve
 has_umfpack = True
 has_cholmod = True
 try:
+    import sksparse
     from sksparse.cholmod import cholesky as cholmod
     from sksparse.cholmod import analyze as cholmod_analyze
 except ImportError:

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -21,6 +21,7 @@ except ImportError:
 has_cholmod = True
 try:
     import sksparse
+    from sksparse.cholmod import cholesky as cholmod
 except ImportError:
     has_cholmod = False
 

